app/controllers/export_controller.rb

```diff
@@ -7,87 +7,170 @@ class ExportController < ApplicationController
-    else
-      # Clear out old receipts and create a new one
-      current_user.export_receipts.where(export_type: type).destroy_all
-      ExportReceipt.create(user_id: current_user.id, export_type: type)
+    # Clear out old receipts and create a new one
+    current_user.export_receipts.where(export_type: export_type).destroy_all
+    ExportReceipt.create(user_id: current_user.id, export_type: export_type)
 
-      # Spawn job to handle export
-      ExportJob.perform_later(current_user.id, type)
+    # Spawn job to handle export
+    config = {
+      user_id: current_user.id,
+      export_type: export_type,
+      format: 'xlsx',
+      filename: "Sara-Alert-Format-#{params[:workflow]&.titleize}",
+      filename_data_type: false,
+      data: {
+        patients: {
+          checked: SARA_ALERT_FORMAT_FIELDS,
+          headers: SARA_ALERT_FORMAT_HEADERS,
+          query: { workflow: params[:workflow] }
+        }
+      }
+    }
 
-      respond_to do |format|
-        format.any { head :ok }
-      end
+    ExportJob.perform_later(config)
+
+    respond_to do |format|
+      format.any { head :ok }
     end
   end
 
   def excel_full_history_patients
-    redirect_to(root_url) && return unless current_user.can_export?
+    # Verify params
+    redirect_to(root_url) && return unless %w[purgeable all].include?(params[:scope])
+
+    export_type = "full_history_#{params[:scope]}".to_sym
+    return if exported_recently?(export_type)
 
-    type = "full_history_#{params[:scope] == 'purgeable' ? 'purgeable' : 'all'}"
+    # Clear out old receipts and create a new one
+    current_user.export_receipts.where(export_type: export_type).destroy_all
+    ExportReceipt.create(user_id: current_user.id, export_type: export_type)
 
-    if current_user.export_receipts.where(export_type: type).where('created_at > ?', 1.hour.ago).exists?
-      render json: { message: 'You have already initiated an export of this type in the last hour. Please try again later.' }.to_json, status: 401
-    else
-      # Clear out old receipts and create a new one
-      current_user.export_receipts.where(export_type: type).destroy_all
-      ExportReceipt.create(user_id: current_user.id, export_type: type)
+    # Spawn job to handle export
+    config = {
+      user_id: current_user.id,
+      export_type: export_type,
+      format: 'xlsx',
+      filename: "Sara-Alert-#{params[:scope] == 'purgeable' ? 'Purge-Eligible' : 'Full'}-Export",
+      filename_data_type: true,
+      separate_files: true,
+      data: {
+        patients: {
+          checked: FULL_HISTORY_PATIENTS_FIELDS,
+          headers: FULL_HISTORY_PATIENTS_HEADERS,
+          query: params[:scope] == 'purgeable' ? { purgeable: true } : {},
+          name: 'Monitorees',
+          tab: 'Monitorees List'
+        },
+        # assessment fields and headers need to be duplicated because they may be modified
+        assessments: {
+          checked: FULL_HISTORY_ASSESSMENTS_FIELDS.dup,
```

Comment:
this is only necessary for the full history exports because these fields and headers are coming from constants defined in import_export whereas for the custom export, they come from query params in which case it doesn't matter if the fields and headers get modified. this also only applies to assessments because the "symptoms_reported" field gets replaced by the actual symptom names and labels

Reply:
Got it, thanks